ui/src/components/Timeline2/Timeline.tsx

```diff
@@ -0,0 +1,147 @@
+import { FC, useReducer, useState, useEffect } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { Button, Intent } from '@blueprintjs/core';
+import { Colors } from '@blueprintjs/colors';
+import { Schema, Entity, Model } from '@alephdata/followthemoney';
+import c from 'classnames';
+import { TimelineRenderer, Layout, Vertex } from './types';
+import { TimelineItem } from './util';
+import {
+  reducer,
+  selectSortedEntities,
+  selectSelectedEntity,
+  selectSelectedVertex,
+} from './state';
+import TimelineEmptyState from './TimelineEmptyState';
+import { TimelineList } from './TimelineList';
+import { TimelineChart } from './TimelineChart';
+import EntityViewer2 from './EntityViewer2';
+import TimelineItemCreateDialog from './TimelineItemCreateDialog';
+
+import './Timeline.scss';
+
+const DEFAULT_COLOR = Colors.BLUE2;
+
+type TimelineProps = {
+  model: Model;
+  entities: Array<Entity>;
+  layout?: Layout;
+  renderer?: TimelineRenderer;
+  writeable?: boolean;
+  fetchEntitySuggestions: (
+    schema: Schema,
+    query: string
+  ) => Promise<Array<Entity>>;
+  onEntityCreateOrUpdate: (entity: Entity) => Promise<Entity>;
+  onEntityRemove: (entity: Entity) => Promise<unknown>;
+  onLayoutUpdate: (layout: Layout) => Promise<unknown>;
+};
+
+const Timeline: FC<TimelineProps> = ({
+  model,
+  entities,
+  layout,
+  renderer,
+  writeable,
+  fetchEntitySuggestions,
+  onEntityCreateOrUpdate,
+  onEntityRemove,
+  onLayoutUpdate,
+}) => {
+  const Renderer = renderer === 'chart' ? TimelineChart : TimelineList;
+
+  const [state, dispatch] = useReducer(reducer, {
+    entities,
+    layout: layout || { vertices: [] },
+    selectedId: null,
+  });
+
+  const [createDialogOpen, setCreateDialogOpen] = useState(false);
+  const toggleCreateDialog = () => setCreateDialogOpen(!createDialogOpen);
+
+  const selectedEntity = selectSelectedEntity(state);
+  const selectedVertex = selectSelectedVertex(state);
+  const sortedEntities = selectSortedEntities(state);
+
+  const items = sortedEntities.map(
+    (entity) => new TimelineItem(entity, layout)
+  );
+
+  useEffect(() => {
+    onLayoutUpdate(state.layout);
+  }, [state.layout, onLayoutUpdate]);
+
+  const createButton = (
+    <Button intent={Intent.PRIMARY} icon="add" onClick={toggleCreateDialog}>
+      <FormattedMessage id="timeline.add_item" defaultMessage="Add item" />
```

Comment:
I thought you added code for i18n here?

Reply:
Yes, the `FormattedMessage` component is a component exposed by `react-intl` it will render a translation if available for the current language or the default message:

https://formatjs.io/docs/react-intl/components#formattedmessage